core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -244,6 +249,25 @@ internal suspend fun processTransientSettingEvents(
             }
         }
 
+        if (BuildCompat.isAtLeastV()) {
+            when (newTransientSettings.flashMode) {
```

Comment:
You should check that the flashMode has changed here before performing this action. It should be gated with a:
```
if (prevTransientSettings.flashMode != newTransientSettings.flashMode) {

}
```